pkg/provisioner/ironic/ironic.go

```diff
@@ -1784,6 +1790,14 @@ func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error
 	return result, nil
 }
 
+func (p *ironicProvisioner) hostName() string {
+	if !p.watchAllNamespaces || p.host.Namespace == p.podNamespace {
+		return p.host.Name
+	}
+
+	return p.host.Namespace + "." + p.host.Name
```

Comment:
Would this break the behaviour during upgrade ? If the nodes are created with an old version of BMO, they only have their name, but then we might be looking for it with namespace + name after the upgrade.

Reply:
I think it depends if we expect to support upgrading from an environment without watchAllNamespaces to one where it's true - there was some discussion about that in the OpenShift case here:

https://github.com/openshift/cluster-baremetal-operator/pull/106#discussion_r576242689

@sadasu @dhellmann did we reach a conclusion on that discussion?

If we expect to support upgrading from a single-namespace to an all-namespaces configuration, I think we'll need to modify the conditional in ValidateManagementAccess where we rename the Ironic host only when `ironicNode.Name == ""` - we could rename whenever the current hostname doesn't match `p.hostName()` there instead?

Reply:
ok, i'll make that change.